### PR TITLE
Palestine Class 1 - Mahmoud Nwiry - JavaScript - 1

### PR DESCRIPTION
**Pull request by Mahmoud-Nwiry**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Mahmoud Nwiry
- Your City: Gaza
- Your Slack Name: Mahmoud Nwiry

# Homework Details

- Module: JavaScript Core 1
- Week: 1

# Notes

- What did you find easy?
- Evrything

- What did you find hard?
- NA

- What do you still not understand?
- NA

- Any other notes?
- NA

### CODE REVIEW

File: mandatory/1-syntax-errors.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
 // There are syntax errors in this code - can you fix it to pass the tests?
 
-function addNumbers(a b c) {
+function addNumbers(a, b, c) {
   return a + b + c;
 }
 
 function introduceMe(name, age)
```
Comment (HassanGhazy) [inline mandatory/1-syntax-errors.js:7]: you miss the curly brackets { }

File: mandatory/2-logic-error.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
 // The syntax for this function is valid but it has an error, find it and fix it.
 
 function trimWord(word) {
-  return wordtrim();
+  return word.trim();
 }
 
 function getStringLength(word) {
-  return "word".length();
+  return word.length();
```
Comment (HassanGhazy) [inline mandatory/2-logic-error.js:8]: word.length not word.length()
length() not a function

**Comment by saraAAZ** [comment]
> **Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_
> 
> # Your Details
> * Your Name:
> * Your City:
> * Your Slack Name:
> 
> # Homework Details
> * Module:
> * Week:
> 
> # Notes
> * What did you find easy?
> * What did you find hard?
> * What do you still not understand?
> * Any other notes?

**Please fill your information in the designated place** 👍✨

**Comment by khaled-Mohamed1** [comment]
nice work

**Comment by nidaa-awawdeh** [comment]
`
function calculateSalesTax(num) {
  let sales = num / 5;
  return num + sales;
}`


 in the code mention 20%  didn't talk about  number five , 
If you use a different equation and give the same result, you must put a comment to the engineer who will read what you did

File: extra/2-piping.js [hunk 3]
```diff
@@ -16,26 +16,35 @@
...
-let badCode =
+
+// Because all the functions are nested and therefore difficult to read
+let badCode = format(multiply(add(startingValue, 10),2));
 
 /* BETTER PRACTICE */
 
-let goodCode =
+// Add the value `startingValue` with the number 10
+let goodCode = add(startingValue, 10);
```
Comment (MohJamal) [inline extra/2-piping.js:41]: Hi Mahmoud,
I have reviewed 2-piping.js file. For me, I don't think reassign the variable with multi functions results is the good practice. but, by creating a variable for each function result till be more readable and easy to be edited.

let add_numbers = add(10, startingValue);
let multiply_numbers = multiply(add_numbers, 2);
let goodCode = format(multiply_numbers);

Keep the good work bro and don't make my comment bother you I am just like you don't know much.

**Pull request closed without merging**